### PR TITLE
[TargetDefinition] Don't include current dependency when using subspecs

### PR DESCRIPTION
**Pull request by kylef**
See https://github.com/CocoaPods/Core/pull/234/files#r27783874

### CODE REVIEW

**Comment by segiddins** [comment]
\c @orta @alloy

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 1]
```diff
@@ -753,17 +753,18 @@ def parse_configuration_whitelist(name, requirements)
       #         If :subspecs is the only key in the hash, the hash
       #         should be destroyed because it confuses Gem::Dependency.
       #
-      # @return [void]
+      # @return [TrueClass,FalseClass] Returns true if we added new subspecs
```
Comment (segiddins) [inline lib/cocoapods-core/podfile/target_definition.rb]: `Boolean`

Comment (segiddins) [inline lib/cocoapods-core/podfile/target_definition.rb]: `Whether new subspecs were added`

File: lib/cocoapods-core/podfile/target_definition.rb [hunk 2]
```diff
@@ -753,17 +753,18 @@ def parse_configuration_whitelist(name, requirements)
...
         options = requirements.last
-        return requirements unless options.is_a?(Hash)
+        return false unless options.is_a?(Hash)
 
         subspecs = options.delete(:subspecs)
         subspecs.each do |ss|
           store_pod("#{name}/#{ss}", *requirements.dup)
         end if subspecs
         requirements.pop if options.empty?
+        !subspecs.nil?
```
Comment (segiddins) [inline lib/cocoapods-core/podfile/target_definition.rb]: can just return `subspecs here`

Comment (kylef) [inline lib/cocoapods-core/podfile/target_definition.rb]: subspecs is an array and not a boolean like this method is expected to return.

Comment (segiddins) [inline lib/cocoapods-core/podfile/target_definition.rb]: Any object can be treated like a Boolean in Ruby. Returning the object is the same as returning whether the object is not nil, it's just less code. 

-Samuel E. Giddins

On Apr 6, 2015, at 5:41 AM, Kyle Fuller <notifications@github.com> wrote:

In lib/cocoapods-core/podfile/target_definition.rb:

>  
>          subspecs = options.delete(:subspecs)
>          subspecs.each do |ss|
>            store_pod("#{name}/#{ss}", *requirements.dup)
>          end if subspecs
>          requirements.pop if options.empty?
> +        !subspecs.nil?
subspecs is an array and not a boolean like this method is expected to return.

Comment (kylef) [inline lib/cocoapods-core/podfile/target_definition.rb]: Not exactly, if it's documented to return a Boolean it should 100% return a boolean.

```ruby
> [].is_a?(TrueClass)
=> false
> [] == true
=> false
```

Comment (segiddins) [inline lib/cocoapods-core/podfile/target_definition.rb]: That's not true by convention, if it documents it returns a Boolean, it just means you can only treat the return value as a Boolean. 

-Samuel E. Giddins

On Apr 6, 2015, at 8:03 AM, Kyle Fuller <notifications@github.com> wrote:

In lib/cocoapods-core/podfile/target_definition.rb:

>  
>          subspecs = options.delete(:subspecs)
>          subspecs.each do |ss|
>            store_pod("#{name}/#{ss}", *requirements.dup)
>          end if subspecs
>          requirements.pop if options.empty?
> +        !subspecs.nil?
Not exactly, if it's documented to return a Boolean it should 100% return a boolean.

> [].is_a?(TrueClass)
=> false
> [] == true
=> false

**Pull request merged**